### PR TITLE
Add failure event for queries that were posted but not submitted

### PR DESCRIPTION
**Pull request by mduggan-starburst**
## Release notes

( ) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
## Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: core/trino-main/src/main/java/io/trino/event/QueryMonitor.java [hunk 1]
```diff
@@ -177,6 +177,55 @@ public void queryCreatedEvent(BasicQueryInfo queryInfo)
                                 Optional::empty)));
     }
 
+    private static QueryStatistics createEmptyStatistics(Duration queuedTime)
```
Comment (losipiuk) [inline core/trino-main/src/main/java/io/trino/event/QueryMonitor.java:180]: unrelated change. Fine on its own but move to separate commit

File: core/trino-main/src/main/java/io/trino/dispatcher/QueuedStatementResource.java [hunk 2]
```diff
@@ -400,15 +440,86 @@ public QueryResults getQueryResults(long token, ExternalUriInfo externalUriInfo)
...
         public void cancel()
         {
-            creationFuture.addListener(() -> dispatchManager.cancelQuery(queryId), directExecutor());
+            cancelled = true;
+            creationFuture.addListener(() -> cancelInternal(), directExecutor());
+        }
+
+        private void cancelInternal()
+        {
+            if (!dispatchManager.cancelQuery(queryId)) {
```
Comment (losipiuk) [inline core/trino-main/src/main/java/io/trino/dispatcher/QueuedStatementResource.java:449]: why can't we have event logic in `dispatchManager.cancelQuery`? 
Is that lack of session or the fact that is used from QueryResource too? Or ... ?

**Comment by losipiuk** [comment]
Looks generally good. @wendigo PTAL too.
Sorry for late review. I did not notice. Please ping me if I am not responsive.

**Comment by hashhar** [comment]
IMO @electrum should also take a look because from my discussion with him in the past this was intentional part of protocol.

https://github.com/trinodb/trino-python-client/pull/394#discussion_r1314923811

**Review by electrum — CHANGES_REQUESTED** [review]
This seems like a bad change, as it will cause bogus events for query resubmission. The query doesn't exist until the client fetches the first next URI.

What problem are you trying to solve? If this is for administrators to debug broken clients, you could add a debug log line for abandoned queries.

**Comment by mduggan-starburst** [comment]
Closed due to conflicts with the Trino protocol

**Pull request closed without merging**